### PR TITLE
khoa's pull request

### PR DESCRIPTION
**Pull request by Omegaeye**
I wasn't able to get to iteration 4. but i hope the rest is ok. i understand it's messy but given the time restraint i tried my best.

### CODE REVIEW

File: lib/character.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Character
+
+  def initialize(role)
+    @role = role
+  end
+
+  def name
+    @role[:name]
```
Comment (robertguturing) [inline lib/character.rb:8]: I think name actor and salary would make more sense to be instance variables of the character class

Comment (Omegaeye) [reply lib/character.rb:8]: i was having a hard time trying to figure out how to access actor and salary through a hash that is the argument. Would it be possible for you to show me a good way to do it?

File: lib/network.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  def initialize(name)
+    @name  = name
+    @shows = []
+  end
+
+  def add_show (show)
+    @shows.push(show)
+  end
+
+  def main_characters
```
Comment (robertguturing) [inline lib/network.rb:15]: The logic behind this is a bit confusing. A character is considered a main character if their name is in all uppercase and they make over 500,000 so I would write the following code.
```
characters = @shows.map { |show| show.characters }.flatten
characters.filter do |character|
  character.salary > 500_000 && character.name == character.name.upcase
end
```

Comment (Omegaeye) [reply lib/network.rb:15]: thank you bob, i wasn't able to figure out an easy way to get the character.name.upcase like you had. thank you for the tip

File: lib/network.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    character = @shows.map{|e|e.characters}.flatten
+    name = character.map{|e| e.name}
+    upcase_name = name.select{|e|e == e.upcase}
+    upcase_name_actor = character.select{|e| e.name[upcase_name.join]}
+    if character.map{|e|e.salary > 500000} && upcase_name
+      upcase_name_actor
+    end
+  end
+
+  def actors_by_show
```
Comment (robertguturing) [inline lib/network.rb:25]: You would need to dynamically add to a hash by looping over the show actors here to create a hash

File: lib/show.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    @creator    = creator
+    @characters = characters
+  end
+
+  def total_salary
+    @characters.sum{|e| e.salary}
+  end
+
+  def highest_paid_actor
+    @characters.max_by{|e|e.salary}.actor
```
Comment (robertguturing) [inline lib/show.rb:16]: Nice use of the max_by enumerable here

**Review by robertguturing — COMMENTED** [review]
Overall some good use of the enumerables, some of the logic is a bit convoluted but no concerns on my end

Score : 2.5

**Pull request closed without merging**